src/main/antora/modules/security/pages/security.adoc

```diff
@@ -776,8 +776,352 @@ public class RestFormAuthCustomStoreIT extends ITBase {
+
+You'll learn how to:
+
+. Define security constraints
+. Use the provided Basic authentication mechanism
+. Use the provided database identity store
+. Create a custom identity store
+. Create a custom identity store handler
+. Use the Jakarta Security `SecurityContext`
+
+=== Write the application
+
+We'll use a slightly modified resource and security constraints compared to the ones we used for the <<Securing an endpoint with Basic authentication>> example.
+
+The REST resource is now as follows:
+
+[source,java]
+----
+@Path("/resource")
+@RequestScoped
+public class Resource {
+
+    @Inject
+    private SecurityContext securityContext;
+
+    @GET
+    @Produces(TEXT_PLAIN)
+    public String getCallerAndRole() {
+        return
+            securityContext.getCallerPrincipal().getName() + " : " +
+            securityContext.isCallerInRole("user") + "," +
+            securityContext.isCallerInRole("caller1") + "," +
+            securityContext.isCallerInRole("caller2");
+    }
+
+}
+----
+
+As can be seen, the difference is quite small; we're now printing out the results of two extra role checks.
+
+
+`web.xml` on its turn looks as follows now:
+
+[source,xml]
+----
+<web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_6_0.xsd"
+         version="6.0">
+
+    <security-constraint>
+        <web-resource-collection>
+            <web-resource-name>protected</web-resource-name>
+            <url-pattern>/rest/*</url-pattern>
+        </web-resource-collection>
+        <auth-constraint>
+            <role-name>user</role-name>
+            <role-name>caller2</role-name>
+        </auth-constraint>
+    </security-constraint>
+    <security-role>
+        <role-name>caller1</role-name>
+    </security-role>
+
+</web-app>
+----
+
+Compared to the example in <<Securing an endpoint with Basic authentication>> we have now added an extra role to the `<auth-constraint>` section. The semantics of that are that a caller needs to have both of these roles in order to be authorised to access the resource under `/rest/*`.
+
+Although it's customary to explicitly declare all roles in the application using `<security-role>`, it's technically not needed. As long as the role name appears in some XML fragment or annotation attribute the Jakarta EE requirement to declare all roles upfront is satisfied. As we can see in the fragment above, the role names "user" and "caller2" already appear in the `<auth-constraint>` section, so they don't *have* to be repeated.
+
+[NOTE]
+The reason it's deemed good practice to list all roles in the `<security-role>` even when not really needed is to have a single place where all roles are listed, instead of them being scattered throughout the application.
```

Comment:
It would help to explicitly say what the alternative is (i.e. using annotations).

Reply:
ok